packages/salesforcedx-vscode-core/src/diagnostics/diagnostics.ts

```diff
@@ -70,12 +81,41 @@ export function handleDiagnosticErrors(
   return errorCollection;
 }
 
-export function getRange(
-  lineNumber: string,
-  columnNumber: string
-): vscode.Range {
-  const ln = Number(lineNumber);
-  const col = Number(columnNumber);
-  const pos = new vscode.Position(ln > 0 ? ln - 1 : 0, col > 0 ? col - 1 : 0);
-  return new vscode.Range(pos, pos);
+export function handleLibraryDiagnostics(
+  deployResult: DeployResult,
+  errorCollection: vscode.DiagnosticCollection
+): vscode.DiagnosticCollection {
+  errorCollection.clear();
+  const diagnosticMap: Map<string, vscode.Diagnostic[]> = new Map();
+
+  deployResult.DeployDetails!.componentFailures.forEach(err => {
```

Comment:
Why can `DeployDetails` possibly be null?

Reply:
I think we missed that on the library side, will be adding it to the work item to address the info that's missing, etc.